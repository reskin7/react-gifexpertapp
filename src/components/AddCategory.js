import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories, setQuantity }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue('');
    }
  };

  const handleQuantity = (event) => {
    setQuantity(event.target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
      <br />
      <br />
      <label htmlFor="quantity">
        Selecciona la cantidad de imagenes que quieres traer {'\n \n'}
      </label>
      <select name="selector" id="quantity" onChange={handleQuantity}>
        <option value={1}>1</option>
        <option value={3}>3</option>
        <option value={5}>5</option>
        <option value={10}>10</option>
      </select>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
