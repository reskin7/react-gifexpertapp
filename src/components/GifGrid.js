import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category, quantity }) => {
  // const [images, setImages] = useState([]);
  const { data: images, loading } = useFetchGifs(category, quantity); // "data: images" es para renombrar...."data" lo voy a poder usar como si fuera una variable de nombre "images"

  // useEffect(() => {
  //   getGifs(category).then(setImages);
  // }, [category]);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>

      {loading && <p className="animate__animated animate__flash">Loading</p>}
      <div className="card-grid">
        {images.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};
