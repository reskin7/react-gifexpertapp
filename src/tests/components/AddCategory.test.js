import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(
    <AddCategory setCategories={setCategories} setQuantity={2} />
  );

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(
      <AddCategory setCategories={setCategories} setQuantity={2} />
    );
  });

  test('should show correctly ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Must change the text input', () => {
    const input = wrapper.find('input');
    const value = 'Hola Mundo';
    input.simulate('change', { target: { value } });

    expect(wrapper.find('p').text().trim()).toBe(value);
  });

  test('should NOT call the setCategories', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call setCategories and clean de textBox', () => {
    const value = 'Overwatch';
    const input = wrapper.find('input');
    // 1. Simular el input change
    input.simulate('change', { target: { value: value } });
    // 2. Simular el submit del form
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    // 3. Debe llamarse el setCategories
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    // 4. Value del input debe estar vacio ('')
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
