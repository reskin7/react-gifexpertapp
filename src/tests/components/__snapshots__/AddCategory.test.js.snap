// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should show correctly  1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <br />
  <label
    htmlFor="quantity"
  >
    Selecciona la cantidad de imagenes que quieres traer 
    
 

  </label>
  <select
    id="quantity"
    name="selector"
    onChange={[Function]}
  >
    <option
      value={1}
    >
      1
    </option>
    <option
      value={3}
    >
      3
    </option>
    <option
      value={5}
    >
      5
    </option>
    <option
      value={10}
    >
      10
    </option>
  </select>
</form>
`;
